barman/clients/cloud_check_wal_archive.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+# © Copyright EnterpriseDB UK Limited 2018-2021
+#
+# This file is part of Barman.
+#
+# Barman is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Barman is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Barman.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+import barman
+from barman.cloud import configure_logging, CloudBackupCatalog
+from barman.cloud_providers import get_cloud_interface
+from barman.exceptions import WalArchiveContentError
+from barman.utils import force_str, check_positive
+from barman.xlog import check_archive_usable
+
+try:
+    import argparse
```

Comment:
Why does the argparse import need to be wrapped in a try-except?

Reply:
This is a good question and I think the answer is "there is no longer any good reason".

The first commit to introduce this was 83e9979a70d8caae5a65ad1b195960cb01ec4cd5 where the `barman-wal-restore.py` script is added. This would have needed to support python 2.6 (and presumably older versions of 3.x) which didn't ship with argparse so there was a genuine possibility it could run without argparse being available, therefore providing a friendly error message was useful.

Now we only support python versions which include argparse I don't think there is a need to do this at all. I'll remove it from this PR and while we're at it I'll update the other scripts too.